terraform/aws/irsa.tf

```diff
@@ -1,20 +1,38 @@
-data "aws_caller_identity" "current" {}
+/*
+  This file provides resources _per hub and role_. Each role is tied to a
+  specific k8s ServiceAccount allowed to assume the role.
+
+  - Role                 - for use by k8s ServiceAccount (user-sa, admin-sa)
+  - Policy               - if extra_iam_policy is declared
+  - RolePolicyAttachment - if extra_iam_policy is declared
+*/
 
+data "aws_caller_identity" "current" {}
 data "aws_partition" "current" {}
 
-resource "aws_iam_role" "irsa_role" {
-  for_each = var.hub_cloud_permissions
-  name     = "${var.cluster_name}-${each.key}"
 
-  assume_role_policy = data.aws_iam_policy_document.irsa_role_assume[each.key].json
+
+locals {
+  hub_role = flatten([
+    for hub, hub_value in var.hub_cloud_permissions : [
+      for role, role_value in hub_value : {
+        // id is conservatively adjusted to not change any previous resource
+        // name set to the hub's name when only "user-sa" roles were around
+        id   = role == "user-sa" ? hub : "${hub}-${role}"
+        hub  = hub
+        role = role
+        data = role_value
```

Comment:
`data` feels too generic here. Can we call it `cloud_permissions` to match what is in the configuration variable name (`hub_cloud_permissions`)?

Reply:
This would also make lines like https://github.com/2i2c-org/infrastructure/pull/3932/files#diff-9f223d043fe1fe9d6f6e3f3a909a3e8d633b2fbbc20f388af72c20069d53cc5aR73 clearer. Just reading that I wasn't sure if `data` was another computed field or not.